modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -783,12 +790,14 @@ CV__DNN_INLINE_NS_BEGIN
          * order is the same as in layersIds
          */
         void getLayerShapes(const MatShape& netInputShape,
+                                    const MatType& netInputType,
```

Comment:
Why type is requires to get only shapes?

Reply:
Since the shape inference and the type inference has the same pipeline, I added the type inference into the shape inference pipeline. So this function probably will be renamed into getLayersShapesAndTypes